src/main/java/com/sweepy/urlService/urlServiceImpl.java

```diff
@@ -0,0 +1,158 @@
+package com.sweepy.urlService;
+
+import com.sweepy.RedisCache.redisService;
+import com.sweepy.converter.base62Converter;
+import com.sweepy.converter.randomConverter;
+import com.sweepy.database.urlTable;
+import com.sweepy.exception.emptyEntry;
+import com.sweepy.exception.nullRequest;
+import com.sweepy.repository.urlRepository;
+import org.apache.commons.validator.UrlValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.transaction.Transactional;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Objects;
+
+@Service
+public class urlServiceImpl implements urlService {
+
+    @Autowired
+    private urlRepository urlRepository;
+    private redisService redisService;
+    long defaultTime = 5;
+
+    private String prefix = "http://www.sweepy.com/";
+    private String protocol = "http://";
+
+
+    private base62Converter base62Converter = new base62Converter();
+    private randomConverter randomConverter = new randomConverter();
+
+
+    public urlServiceImpl(urlRepository urlRepository, redisService redisService) {
+        this.urlRepository = urlRepository;
+        this.redisService = redisService;
+    }
+
+    @Override
+    @Transactional
+    public String longToShort(String longUrl, String method) {
+        String shortUrl = "";
+        urlTable entry = null;
+
+        String shortInRedis = (String) redisService.get(longUrl);
+        if (shortInRedis == null) {
+            entry = urlRepository.findByLongUrl(longUrl);
+
+            if (!isValid(protocol + longUrl)) {
+                return "You must enter a valid long Url.";
+            } else if (entry == null || !Objects.equals(entry.getMethod(), method)) {
+                if (Objects.equals(method, "base62")) {
+                    shortUrl = base62Converter.encode(longUrl);
```

Comment:
should use counter(sequence) to generate base62 code. In  base62convetr,  it just sums up long Url and converts the sum to base62.

Reply:
addressed